Error in ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27039)

### Describe the bug

While rolling consummables (Food, Water, Arrow or Torches), the following error is thrown : 
```
forbidden-lands.js:44 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'result')
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27039)
rollConsumable @ forbidden-lands.js:44
```

Which prevent the consummable dice to be decreased if the roll is a 1 or a 2.

### Reproduction

1. Open an existing character sheet or create a new one
2. Set a consummable to any value, for example Food at D10
3. Open the console (F12)
4. Roll for the consummables
5. Notice the error
6. If the roll result was a 1 or a 2, confirm the dice value is not decreased

### System Info

```shell
Foundry Version: 12.330
Forbidden Lands System Version: 12.0.2
```


### Logs

```
foundry-esm.js:4621 Error: The async option for Roll#evaluate has been removed. Use Roll#evaluateSync for synchronous roll evaluation.
    at Object.logCompatibilityWarning (foundry-esm.js:4609:19)
    at FBLRoll.evaluate (foundry-esm.js:63460:23)
    at FBLRoll.roll (foundry-esm.js:63641:19)
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:26967)
    at HTMLButtonElement.<anonymous> (forbidden-lands.js:44:25714)
    at HTMLButtonElement.dispatch (jquery.min.js:2:40035)
    at v.handle (jquery.min.js:2:38006)
logCompatibilityWarning @ foundry-esm.js:4621
evaluate @ foundry-esm.js:63460
roll @ foundry-esm.js:63641
rollConsumable @ forbidden-lands.js:44
(anonymous) @ forbidden-lands.js:44
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
foundry-esm.js:4621 Error: CONST.CHAT_MESSAGE_TYPES is deprecated in favor of CONST.CHAT_MESSAGE_STYLES because the ChatMessage#type field has been renamed to ChatMessage#style
Deprecated since Version 12
Backwards-compatible support will be removed in Version 14
    at Object.logCompatibilityWarning (foundry-esm.js:4609:19)
    at Object.get (foundry-esm.js:2492:21)
    at FBLRoll.toMessage (forbidden-lands.js:17:42210)
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27007)
logCompatibilityWarning @ foundry-esm.js:4621
get @ foundry-esm.js:2492
toMessage @ forbidden-lands.js:17
rollConsumable @ forbidden-lands.js:44
await in rollConsumable (async)
(anonymous) @ forbidden-lands.js:44
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
foundry-esm.js:4621 Error: CONST.CHAT_MESSAGE_STYLES.ROLL is deprecated in favor of defining rolls directly in ChatMessage#rolls
Deprecated since Version 12
Backwards-compatible support will be removed in Version 14
    at Object.logCompatibilityWarning (foundry-esm.js:4609:19)
    at Proxy.get (foundry-esm.js:2505:23)
    at Reflect.get (<anonymous>)
    at Object.get (foundry-esm.js:2493:22)
    at FBLRoll.toMessage (forbidden-lands.js:17:42210)
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27007)
logCompatibilityWarning @ foundry-esm.js:4621
get @ foundry-esm.js:2505
get @ foundry-esm.js:2493
toMessage @ forbidden-lands.js:17
rollConsumable @ forbidden-lands.js:44
await in rollConsumable (async)
(anonymous) @ forbidden-lands.js:44
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
foundry-esm.js:4621 Error: CONST.CHAT_MESSAGE_TYPES is deprecated in favor of CONST.CHAT_MESSAGE_STYLES because the ChatMessage#type field has been renamed to ChatMessage#style
Deprecated since Version 12
Backwards-compatible support will be removed in Version 14
    at Object.logCompatibilityWarning (foundry-esm.js:4609:19)
    at Object.get (foundry-esm.js:2492:21)
    at FBLRoll.toMessage (forbidden-lands.js:17:23499)
    at FBLRoll.toMessage (forbidden-lands.js:17:42241)
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27007)
logCompatibilityWarning @ foundry-esm.js:4621
get @ foundry-esm.js:2492
toMessage @ forbidden-lands.js:17
toMessage @ forbidden-lands.js:17
rollConsumable @ forbidden-lands.js:44
await in rollConsumable (async)
(anonymous) @ forbidden-lands.js:44
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
foundry-esm.js:4621 Error: CONST.CHAT_MESSAGE_STYLES.ROLL is deprecated in favor of defining rolls directly in ChatMessage#rolls
Deprecated since Version 12
Backwards-compatible support will be removed in Version 14
    at Object.logCompatibilityWarning (foundry-esm.js:4609:19)
    at Proxy.get (foundry-esm.js:2505:23)
    at Reflect.get (<anonymous>)
    at Object.get (foundry-esm.js:2493:22)
    at FBLRoll.toMessage (forbidden-lands.js:17:23499)
    at FBLRoll.toMessage (forbidden-lands.js:17:42241)
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27007)
logCompatibilityWarning @ foundry-esm.js:4621
get @ foundry-esm.js:2505
get @ foundry-esm.js:2493
toMessage @ forbidden-lands.js:17
toMessage @ forbidden-lands.js:17
rollConsumable @ forbidden-lands.js:44
await in rollConsumable (async)
(anonymous) @ forbidden-lands.js:44
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
forbidden-lands.js:44 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'result')
    at ForbiddenLandsCharacterSheet.rollConsumable (forbidden-lands.js:44:27039)
```

### Validations

- [X] Done a clean install of the system.
- [X] Disabled all modules.
- [X] Checked if the issue is present in a new world instance (create a new world in foundry using the Forbidden Lands system).
- [X] Not a bug report about the content modules. Reports regarding the content modules go [here instead](https://github.com/aMediocreDad/fbl-content-modules/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).
- [X] Not a "How To"-question (You can ask questions here: [Discussions](https://github.com/fvtt-fria-ligan/forbidden-lands-foundry-vtt/discussions)).

--- a/.changeset/fuzzy-mangos-rhyme.md
+++ b/.changeset/fuzzy-mangos-rhyme.md
@@ -1,0 +1,5 @@
+---
+"forbidden-lands": patch
+---
+
+Consumable roll not correctly decreasing consumables

--- a/lang/en.json
+++ b/lang/en.json
@@ -353,6 +353,7 @@
 	"MONSTER.TYPE.MOUNT": "Mount",
 	"MOVEMENT": "Movement",
 	"NONE": "None",
+	"NOTIFY.CONSUMABLE_USED": "Consumable used!",
 	"NOTIFY.YOUR_ITEM_BROKE": "Your item broke!",
 	"NOTIFY.YOU_ARE_BROKEN": "You are broken!",
 	"OTHER": "OTHER",

--- a/src/actor/character/character-sheet.js
+++ b/src/actor/character/character-sheet.js
@@ -127,7 +127,7 @@ export class ForbiddenLandsCharacterSheet extends ForbiddenLandsActorSheet {
 		await roll.roll({ async: true });
 		const message = await roll.toMessage();
 		if (
-			Number(message.roll.result) <=
+			Number(message.rolls[0].result) <=
 			(game.settings.get("forbidden-lands", "autoDecreaseConsumable") || 0)
 		) {
 			FBLRollHandler.decreaseConsumable(message.id);

--- a/src/components/roll-engine/engine.js
+++ b/src/components/roll-engine/engine.js
@@ -704,15 +704,18 @@ export class FBLRollHandler extends FormApplication {
 	 */
 	static async decreaseConsumable(messageId) {
 		let {
-			data: { speaker },
-			roll: {
-				options: { consumable },
-			},
+			speaker,
+			rolls: [
+				{
+					options: { consumable },
+				},
+			],
 		} = game.messages.get(messageId);
 
 		speaker = this.getSpeaker(speaker);
 		if (!speaker)
 			return console.error("Could not decrease consumable: No actor found.");
+		ui.notifications.info("NOTIFY.CONSUMABLE_USED", { localize: true });
 
 		const currentValue = speaker?.consumables[consumable]?.value;
 		const newValue = Math.max(currentValue - 1, 0);

